src/validation/obfl.rng

```diff
@@ -1513,6 +1506,9 @@
     <attribute name="ref-id">
       <data type="IDREF"/>
     </attribute>
+    <ref name="attlist-number-format"/>
+  </define>
+  <define name="attlist-number-format" combine="interleave">
```

Comment:
It's better to remove the `optional` out of the `define`. The element that references this attribute can then decide whether it is optional or not. Also don't call it "attlist-number-format" but for example "number-format-att". "attlist-number-format" would mean "the attributes of the `number-format` element". I think you can also drop the `combine`.

Reply:
I would like to keep it as it is, because:
- the current file does not have anything that ends with "-att" so what you suggest breaks the way the current schema is defined
- there are several examples where the `optional` is in the `define`. Even examples with nothing but a single `define`. See, e.g., `attlist-volume-transition`, `attlist-on-toc-start`
- all the attribute definitions, even those with 0 or 1 attributes, have a `combine`

I prefer not to break the way the RNG is currently constructed. Therefore I prefer not to apply these suggested changes.

Reply:
You are breaking the meaning of the "attlist-foo" pattern that is currently used in the file.

To me the `combine="interleave"` only makes sense when it's a list of attributes that you might want to extend with more attributes. This is true for the existing "attlist" definitions but it's not true in this case.

The examples where the `optional` is inside the `define` is when we're defining an attribute list of a certain element. In this case you are defining an attribute that you reference in those attribute lists (the "attlist-current-page" and "attlist-evaluate" definitions).

Reply:
Ok, I'm convinced. :-)
I have made the changes.